splice_machine/src/test/java/com/splicemachine/derby/test/framework/SpliceWatcher.java

```diff
@@ -160,21 +173,61 @@ public PreparedStatement prepareStatement(String sql) throws SQLException {
         return ps;
     }
 
+    /**
+     * Try closing connections gracefully from different threads, if that takes long it could be some queries are stuck,
+     * so abort the connections forcefully instead
+     */
     private void closeConnections() {
+        int numThreads = connections.size() + 2; // two extra for the .abort() call
+        ExecutorService executor = Executors.newFixedThreadPool(numThreads,
+                new ThreadFactoryBuilder().setDaemon(true).setNameFormat("connection-close-%d").build());
+        CompletionService<Void> completionService =
+                new ExecutorCompletionService<>(executor);
         try {
-            for (Connection connection : connections) {
-                if (connection != null && !connection.isClosed()) {
-                    try {
-                        connection.close();
-                    } catch (SQLException e) {
-                        connection.rollback();
+            connections.stream().forEach(connection -> completionService.submit(() -> {
+                try {
+                    if (connection != null && !connection.isClosed())
                         connection.close();
+                } catch (SQLException e) {
```

Comment:
Does it make sense to at least log this exception?

Reply:
Good point, made the change.